src/NHibernate/Transaction/AdoTransaction.cs

```diff
@@ -190,20 +190,19 @@ public void Commit()
 
 				log.Debug("Start Commit");
 
+				//Hibernate calls flush from inside BeforeTransactionCompletion
+				if (session.FlushMode != FlushMode.Manual)
+					session.Flush();
+				session.BeforeTransactionCompletion(this);
```

Comment:
Why not moving it inside? It does not look to be called in some paths not doing the flush too, so it should be the same, while reducing code duplication.